### ISSUE TITLE
TypeError: Error #1006: isNaN is not a function on flash target when calling Std.parseInt()

### ISSUE THREAD

**Issue by joshtynjala** [HaxeFoundation/haxe#11383]
The change in commit 45f547f0797f20447c89dee5044f834482ee776d seems to have broken `Std.parseInt()` on the flash target, in some situations. In one of my projects, I get the following exception with Haxe 4.3 that I did not get in previous versions.

> TypeError: Error #1006: isNaN is not a function.
>	at Std$/parseInt()[/usr/local/lib/haxe/std/flash/_std/Std.hx:54]

Interestingly, I just tried using `Std.parseInt()` in a new, empty project, and it worked fine.

As best I can tell, it seems to be failing specifically in a project compiled from Haxe into a _.swc_ and then that _.swc_ is used in an AS3 project. Perhaps it is trying to use flash's `Math` class instead of the Haxe `Math` class (just a wild guess without knowing much about how Haxe implements SWF bytecode)?

If I modify Std.hx locally to call the Flash's global `isNaN()` again instead of `Math.isNaN()`, it works correctly again, like Haxe < 4.3.

```haxe
if (untyped __global__["isNaN"](x))
```

**Comment by Simn** [comment]
This sounds like `Math.__init__` is never run. IIRC you have to call [this thing](https://github.com/HaxeFoundation/haxe/blob/development/std/flash/Boot.hx#L31) from somewhere.

**Comment by joshtynjala** [comment]
Thanks!

I've gone for multiple years without calling anything special to initialize Haxe SWCs, and I assumed it must have been something required only in the past. I guess I just got lucky until now.

However, I confirm that adding the following code during startup of my AS3 project fixed the issue.

```
haxe.initSwc(null);
```

I passed null because my root display object is a `Sprite` instead of a `MovieClip`, so I didn't have an actual `MovieClip` available.
